syntax = "proto3";

package replicateddatabase;

message ArgsVote {
    uint32 term = 1;
    uint32 candidateId = 2;
    int32 lastLogIndex = 3;
    int32 lastLogTerm = 4;
} 

message RetVote {
    bool voteGranted = 1;
    uint32 term = 2;
} 

message ArgsLog {
    string message = 1;
}

message RetLog {
    bool response = 1;
}

service RaftBook {
    rpc RequestVote(ArgsVote) returns (RetVote) {}
}
